cmd/juju/cloud/add_test.go

```diff
@@ -499,7 +501,47 @@ func (*addSuite) TestInteractiveAddCloud_PromptForNameIsCorrect(c *gc.C) {
 	// Running the command will return an error because we only give
 	// enough input to get to the prompt we care about checking. This
 	// test ignores this error.
-	command.Run(ctx)
+	err := command.Run(ctx)
+	c.Assert(errors.Cause(err), gc.Equals, io.EOF)
 
 	c.Check(out.String(), gc.Matches, "(?s).+Enter a name for your manual cloud: .*")
 }
+
+func (*addSuite) TestSpecifyingCloudFileThroughFlag_CorrectlySetsMemberVar(c *gc.C) {
+	command := cloud.NewAddCloudCommand(nil)
+	runCmd := func() {
+		testing.RunCommand(c, command, "garage-maas", "-f", "fake.yaml")
+	}
+	c.Assert(runCmd, gc.PanicMatches, "runtime error: invalid memory address or nil pointer dereference")
+	c.Check(command.CloudFile, gc.Equals, "fake.yaml")
+}
+
+func (*addSuite) TestSpecifyingCloudFileThroughFlagAndArgument_Errors(c *gc.C) {
+	command := cloud.NewAddCloudCommand(nil)
+	_, err := testing.RunCommand(c, command, "garage-maas", "-f", "fake.yaml", "foo.yaml")
```

Comment:
can we do this same test with the two filenames the same?  I think that should be an error as well.